static/js/components/dashboard/courses/StatusMessages.js

```diff
@@ -95,6 +95,7 @@ export const calculateMessages = (props: CalculateMessagesProps) => {
   const passedExam = hasPassingExamGrade(course)
   const failedExam = hasFailingExamGrade(course)
   const paymentDueDate = moment(firstRun.course_upgrade_deadline)
+  const dueDate = paymentDueDate.format(COURSE_CARD_FORMAT)
```

Comment:
@annagav do we have always `firstRun.course_upgrade_deadline` set?

Reply:
We should, but it would be good to fail gracefully if it's missing.